uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -451,28 +451,41 @@ public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDa
    *   And array of values from the Leave Request Status OptionGroup
    * @param bool $publicHolidaysOnly
    *   When true, will get the balance change only for the Public Holiday Leave Requests
+   * @param bool $expiredOnly
+   *   When true, will only count the balance change for expired days
    *
    * @return array
    */
   public static function getBalanceChangeByAbsenceType(
     $contactID,
     $periodID,
     $leaveRequestStatus = [],
-    $publicHolidaysOnly = false
+    $publicHolidaysOnly = false,
+    $expiredOnly = false
```

Comment:
As much as I don't like this too, but looking at it from the frontend perspective, It think it will be fine to go ahead with the first option you suggested earlier.

`On the civicrm_api3_leave_request_getbalancechangebyabsencetype() function (yeah, directly on the API function. I think this is ok because this is the only place the method will be used anyway and it also avoids duplication of tests between BAO and API)`